scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,168 @@
+            else:
+                fails += 1
+        elif instance.status == 'skipped':
+            skips += 1
+        else:
+            passes += 1
+
+    tree = ET.parse(filename)
+    assert int(tree.findall('testsuite')[0].attrib["skip"]) == int(skips)
+    assert int(tree.findall('testsuite')[0].attrib["failures"]) == int(fails)
+    assert int(tree.findall('testsuite')[0].attrib["errors"]) == int(errors)
+    assert int(tree.findall('testsuite')[0].attrib["tests"]) == int(passes+fails+skips+errors)
+
+    for index in range(0, len(class_testsuite.instances)):
+        if len(list(tree.findall('testsuite')[0][index])) != 0:
+            if tree.findall('testsuite')[0][index][0].attrib["type"] == "failure":
+                assert tree.findall('testsuite')[0][index].attrib["name"] == \
+                       (inst1.testcase.name)
+            elif tree.findall('testsuite')[0][index][0].attrib["type"] == "skipped":
+                assert tree.findall('testsuite')[0][index].attrib["name"] == \
+                       (inst2.testcase.name)
+    os.remove(filename)
+
+def test_compare_metrics(class_testsuite, test_data, instances_fixture, caplog):
+    """ Testing compare_metrics function of Testsuite class in sanitycheck
+    Test 1: Error message is raised if file sanitycheck.csv file doesnt exist
+    Test 2: Assert on compare_metrics results for expected values"""
+    class_testsuite.instances = instances_fixture
+    for instance in class_testsuite.instances.values():
+        instance.metrics["ram_size"] = 5
+        instance.metrics["rom_size"] = 9
+    filename_not_exist = test_data + "sanitycheck_file_not_exist.csv"
+    class_testsuite.compare_metrics(filename_not_exist)
+    assert "Cannot compare metrics, " + filename_not_exist + " not found" in caplog.text
+
+    filename = test_data + "sanitycheck.csv"
+    results = class_testsuite.compare_metrics(filename)
+    for instance in class_testsuite.instances.values():
+        for res in results:
+            assert res[0].platform.name == instance.platform.name
+            if (res[0].platform.name == instance.platform.name) and \
+               (res[0].testcase.name == instance.testcase.name):
+                if res[1] == "ram_size":
+                    assert res[2] == instance.metrics["ram_size"]
+                elif res[1] == "rom_size":
+                    assert res[2] == instance.metrics["rom_size"]
+
+def test_target_report(class_testsuite, instances_fixture, tmpdir_factory):
+    """ Testing target_report function of Testsuite class in sanitycheck
+    Test: Assert xunit_report function is called from target_report function"""
+    class_testsuite.instances = instances_fixture
+    outdir = tmpdir_factory.mktemp("tmp")
+    class_testsuite.xunit_report = MagicMock(side_effect=class_testsuite.xunit_report)
+    class_testsuite.target_report(outdir, "abc", append=False)
+    assert class_testsuite.instances
+    class_testsuite.xunit_report.assert_called()
```

Comment:
I am not sure what you are testing here, testing that a function was called is not verifying anything, what about the output itself, does produce the output we expect.

Reply:
This is because there is no output from target report function. Neither it returns or modifies anything, except calling the xunit report function.